sled-agent/src/services.rs

```diff
@@ -512,21 +512,35 @@ impl ServiceManager {
     // These will fail if the disks aren't attached.
     // Should we have a retry loop here? Kinda like we have with the switch
     // / NTP zone?
-    pub async fn load_services(&self) -> Result<(), Error> {
+    pub async fn load_services(
+        &self,
+        boundary_switch_addrs: HashSet<Ipv6Addr>,
+    ) -> Result<(), Error> {
         let log = &self.inner.log;
         let ledger_paths = self.all_service_ledgers().await;
         info!(log, "Loading services from: {ledger_paths:?}");
 
         let mut existing_zones = self.inner.zones.lock().await;
-        let Some(ledger) = Ledger::<AllZoneRequests>::new(
+        let Some(mut ledger) = Ledger::<AllZoneRequests>::new(
```

Comment:
I'm kinda confused how this works when used in the `services_put` API.

Suppose Nexus calls that endpoint with a set of services:

https://github.com/oxidecomputer/omicron/blob/fa3b74bee6033261c9f83c1a5cdf7e1e79c4b332/sled-agent/src/sled_agent.rs#L611-L639

When we're in the body of that function:

https://github.com/oxidecomputer/omicron/blob/fa3b74bee6033261c9f83c1a5cdf7e1e79c4b332/sled-agent/src/services.rs#L2445

It seems like we would be missing this critical boundary services information, no?

Reply:
Yeah good catch. This definitely does not work as written. I'm still not super familiar with all the pieces here, but it looks like this is the only other place we persist `AllZoneRequests` to a ledger.

Reply:
I believe I resolved this in [629065b](https://github.com/oxidecomputer/omicron/pull/3729/commits/629065b70460ed497b4661158de9f8492445c1bb)